SmartDeviceLinkTests/SDLStreamingMediaLifecycleManagerSpec.m

```diff
@@ -7,59 +7,87 @@
         
-        [NSThread sleepForTimeInterval:0.1];
+        [NSThread sleepForTimeInterval:0.3];
     };
     
     beforeEach(^{
-        someVideoEncoderSettings = @{
+        testConfiguration.customVideoEncoderSettings = @{
                                      (__bridge NSString *)kVTCompressionPropertyKey_ExpectedFrameRate : @1
                                      };
+        testConfiguration.dataSource = testDataSource;
         someBackgroundTitleString = @"Open Test App";
-        streamingLifecycleManager = [[SDLStreamingMediaLifecycleManager alloc] initWithEncryption:streamingEncryptionFlag videoEncoderSettings:someVideoEncoderSettings];
+        testConnectionManager = [[TestConnectionManager alloc] init];
+        streamingLifecycleManager = [[SDLStreamingMediaLifecycleManager alloc] initWithConnectionManager:testConnectionManager configuration:testConfiguration];
     });
     
     it(@"should initialize properties", ^{
         expect(streamingLifecycleManager.touchManager).toNot(beNil());
-        expect(@(streamingLifecycleManager.isVideoStreamingSupported)).to(equal(@NO));
-        expect(@(streamingLifecycleManager.isAudioStreamingSupported)).to(equal(@NO));
+        expect(@(streamingLifecycleManager.isStreamingSupported)).to(equal(@NO));
         expect(@(streamingLifecycleManager.isVideoConnected)).to(equal(@NO));
         expect(@(streamingLifecycleManager.isAudioConnected)).to(equal(@NO));
         expect(@(streamingLifecycleManager.isVideoEncrypted)).to(equal(@NO));
         expect(@(streamingLifecycleManager.isAudioEncrypted)).to(equal(@NO));
         expect(@(streamingLifecycleManager.isVideoStreamingPaused)).to(equal(@YES));
         expect(@(CGSizeEqualToSize(streamingLifecycleManager.screenSize, CGSizeZero))).to(equal(@YES));
         expect(@(streamingLifecycleManager.pixelBufferPool == NULL)).to(equal(@YES));
-        expect(@(streamingLifecycleManager.requestedEncryptionType)).to(equal(@(streamingEncryptionFlag)));
+        expect(@(streamingLifecycleManager.requestedEncryptionType)).to(equal(@(SDLStreamingEncryptionFlagNone)));
         expect(streamingLifecycleManager.currentAppState).to(equal(SDLAppStateActive));
         expect(streamingLifecycleManager.currentAudioStreamState).to(equal(SDLAudioStreamStateStopped));
         expect(streamingLifecycleManager.currentVideoStreamState).to(equal(SDLVideoStreamStateStopped));
+        expect(streamingLifecycleManager.videoFormat).to(beNil());
+        expect(streamingLifecycleManager.dataSource).to(equal(testDataSource));
+        expect(streamingLifecycleManager.supportedFormats).to(haveCount(2));
+        expect(streamingLifecycleManager.preferredFormats).to(beNil());
+        expect(streamingLifecycleManager.preferredResolutions).to(beNil());
+        expect(streamingLifecycleManager.preferredFormatIndex).to(equal(0));
+        expect(streamingLifecycleManager.preferredResolutionIndex).to(equal(0));
     });
     
     describe(@"when started", ^{
```

Comment:
If you put a breakpoint inside the `startWithProtocol:protocol:completionHandler` method you can see that completion handler is never called

```
 [streamingLifecycleManager startWithProtocol:protocolMock completionHandler:^(BOOL success, NSError * _Nullable error) {
                readyHandlerSuccess = success;
                readyHandlerError = error;
            }];
```